modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemsXlsMVCResourceCommand.java

```diff
@@ -130,108 +235,90 @@ protected void doServeResource(
-
-			workbookBuilder.cell(
-				contentDashboardItem.getTitle(locale)
-			).cell(
-				contentDashboardItem.getUserName()
-			).cell(
-				contentDashboardItem.getTypeLabel(locale)
-			).cell(
-				Optional.ofNullable(
-					contentDashboardItem.getContentDashboardItemSubtype()
-				).map(
-					contentDashboardItemSubtype ->
-						contentDashboardItemSubtype.getLabel(locale)
-				).orElse(
-					StringPool.BLANK
-				)
-			).cell(
-				contentDashboardItem.getScopeName(locale)
-			).cell(
-				() -> {
-					List<ContentDashboardItem.Version> versions =
-						contentDashboardItem.getVersions(locale);
-
-					ContentDashboardItem.Version version = versions.get(0);
-
-					return version.getLabel();
+			for (Document document : documents) {
+				ContentDashboardItem<?> contentDashboardItem =
+					_toContentDashboardItem(document);
+
+				if (contentDashboardItem != null) {
+					workbookBuilder.row();
+
+					_addWorkbookCell(
+						locale, workbookBuilder, contentDashboardItem);
 				}
-			).cell(
-				StringUtils.joinWith(
-					", ",
-					ListUtil.toList(
-						contentDashboardItem.getAssetCategories(),
-						assetCategory -> assetCategory.getTitle(locale)))
-			).cell(
-				StringUtils.joinWith(
-					", ",
-					ListUtil.toList(
-						contentDashboardItem.getAssetTags(),
-						AssetTagModel::getName))
-			).cell(
-				_toString(contentDashboardItem.getModifiedDate())
-			).cell(
-				contentDashboardItem.getDescription(locale)
-			);
-
-			Map<String, Object> specificInformation =
-				contentDashboardItem.getSpecificInformation(locale);
-
-			workbookBuilder.cell(_toString(specificInformation, "extension"));
-
-			if (contentDashboardItem.getClipboard() != null) {
-				ContentDashboardItem.Clipboard clipboard =
-					contentDashboardItem.getClipboard();
-
-				workbookBuilder.cell(_toString(clipboard.getName()));
 			}
 
-			workbookBuilder.cell(
-				_toString(specificInformation, "size")
-			).cell(
-				_toString(specificInformation, "display-date")
-			).cell(
-				_toString(contentDashboardItem.getCreateDate())
-			);
+			if (documents.size() < indexSearchLimit) {
+				break;
+			}
+
+			start = start + indexSearchLimit;
+		}
+	}
 
-			List<Locale> locales = contentDashboardItem.getAvailableLocales();
+	private SearchResponse _getSearchResponse(
+		int end, ResourceRequest resourceRequest, int start) {
+
+		SearchContext searchContext = new ContentDashboardSearchContextBuilder(
+			_portal.getHttpServletRequest(resourceRequest),
+			_assetCategoryLocalService, _assetVocabularyLocalService
+		).withEnd(
+			end
+		).withStart(
+			start
+		).build();
+
+		searchContext.setSorts(
+			new Sort(Field.MODIFIED_DATE, Sort.LONG_TYPE, false),
+			new Sort(Field.CLASS_NAME_ID, Sort.LONG_TYPE, false),
+			new Sort(Field.CLASS_PK, Sort.LONG_TYPE, false));
+
+		SearchRequest searchRequest =
+			_contentDashboardSearchRequestBuilderFactory.builder(
+				searchContext
+			).build();
+
+		return _searcher.search(searchRequest);
```

Comment:
Maybe Brian prefer to make searchRequest inline?

Reply:
definetly, will change this :)